File: Python/Algorithms/684.py
class Solution:
    """ Note: for UnionFind structure, see Structures/UnionFind/uf.py. """
    def findRedundantConnection(self, edges: List[List[int]]) -> List[int]:
        # Yes, goes from 0-n+1 but we don't use 0.
        uf = UnionFind(range(len(edges)+1))

        last = None
        for (u, v) in edges:
            if uf.find(u) == uf.find(v):
                last = [u, v]
            else:
                uf.union(u, v)
        return last

File: Javascript/Algorithms/684.js
// Note: For UnionFind structure, see Structures/UnionFind/uf.js
var findRedundantConnection = function(edges) {
  let uf = new UnionFind(1, edges.length);

  let last = [];
  edges.forEach(([a, b]) => {
    if (uf.find(a) === uf.find(b))
      last = [a, b];
    else
      uf.union(a, b)
  });
  return last;
};